### PR TITLE
AnnouncementManager: make a copy of file item

### PR DESCRIPTION
**Pull request by FernetMenta**
Make a copy of CFileItem. The original may be changed elsewhere.

@MilhouseVH could you test if this fixes the crash?

### CODE REVIEW

**Comment by MilhouseVH** [comment]
@FernetMenta: that seems to have fixed the crash, many thanks!

**Comment by FernetMenta** [comment]
@MilhouseVH thanks
jenkins build this please

File: xbmc/interfaces/AnnouncementManager.cpp [hunk 1]
```diff
@@ -134,9 +136,9 @@ void CAnnouncementManager::DoAnnounce(AnnouncementFlag flag, const char *sender,
     announcers[i]->Announce(flag, sender, message, data);
 }
 
-void CAnnouncementManager::DoAnnounce(AnnouncementFlag flag, const char *sender, const char *message, CFileItemPtr item, CVariant &data)
+void CAnnouncementManager::DoAnnounce(AnnouncementFlag flag, const char *sender, const char *message, const CFileItemPtr item, const CVariant &data)
 {
-  if (!item.get())
+  if (item == nullptr)
```
Comment (ksooo) [inline xbmc/interfaces/AnnouncementManager.cpp]: @paxxi Just curious how this works internally? what does the compiler make out of this?

Comment (FernetMenta) [inline xbmc/interfaces/AnnouncementManager.cpp]: shared_ptr has overloaded operators

Comment (ksooo) [inline xbmc/interfaces/AnnouncementManager.cpp]: of course, but which of the conversion operators is choosen in this case? is it well defined and portable?

Comment (Montellese) [inline xbmc/interfaces/AnnouncementManager.cpp]: It uses
```
template <class T>
    bool operator==(const shared_ptr<T>& lhs, std::nullptr_t rhs);
```
which should be part of the standard and therefore is portable.

File: xbmc/interfaces/AnnouncementManager.h [hunk 2]
```diff
@@ -46,14 +46,14 @@ namespace ANNOUNCEMENT
     void RemoveAnnouncer(IAnnouncer *listener);
 
     void Announce(AnnouncementFlag flag, const char *sender, const char *message);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, CVariant &data);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, CFileItemPtr item);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, CFileItemPtr item, CVariant &data);
+    void Announce(AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data);
+    void Announce(AnnouncementFlag flag, const char *sender, const char *message, const CFileItemPtr item);
```
Comment (tamland) [inline xbmc/interfaces/AnnouncementManager.h]: `item` is a value parameter so adding const in this context is pretty pointless.

I would check if it's possible to make the parameter `shared_ptr<const CFileItem>` instead, as it should have been from the beginning. If it requires too much changes, then fallback to copying.

File: xbmc/interfaces/AnnouncementManager.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/interfaces/AnnouncementManager.cpp]: thx @Montellese

**Comment by FernetMenta** [comment]
I dropped the const from the value parameter, thank @tamland 
Don't have more time to try your other idea. Want to get this in asap because it fixes a crash.

jenkins build this please

**Pull request merged**